app/models/order_item.rb

```diff
@@ -0,0 +1,17 @@
+class OrderItem < ApplicationRecord
+  belongs_to :product
+  belongs_to :order
+
+  validates_numericality_of :quantity, greater_than: 0
+  
+  def total
+    (self.quantity * self.product.price)
+  end
+
+  def increase_quantity(quantity)
+    existing_quantity = self.quantity
+    new_quantity = existing_quantity + quantity
+
+    return self.update(quantity: new_quantity)
```

Comment:
This is where it'd be important to check the quantity against the quantity in stock.